streams/client.go

```diff
@@ -90,15 +93,33 @@ func (client *client) publishEvents(events []publisher.Event) ([]publisher.Event
 		logp.Debug("kinesis", "sent %d records: %v", len(records), records)
 		observer.Dropped(dropped)
 		observer.Acked(len(okEvents))
+		if len(records) == 0 {
+			logp.Info("kinesis", "No records were mapped")
+			return nil, nil
+		}
 	}
 	logp.Debug("kinesis", "mapped to records: %v", records)
 	res, err := client.putKinesisRecords(records)
 	failed := collectFailedEvents(res, events)
-	if err != nil && len(failed) == 0 {
-		failed = events
+	if len(failed) == 0 {
```

Comment:
What not > 0

Reply:
Because if we get API throttled, for example, then this won't show up as individual Record errors. The call itself will fail. So len would be 0.